src/httpation_request.erl

```diff
@@ -63,3 +71,143 @@ new(String) when is_list(String) ->
     {ok, #http_request{}};
 new(_) ->
     {error, not_a_string}.
+
+
+-spec method(Request) -> Method
+    when Request :: httpation:request(),
+         Method  :: method().
+    %% @doc
+    %% Returns HTTP method of the given request
+
+method(#http_request{method = Method}) ->
+    Method.
+
+
+-spec method(Request, Method) -> NewRequest
```

Comment:
Don't pass the Record in first, pass it in last.
Almost everything in the standard library puts the object of the operation last in the argument list, and the operands first.

Following this, we want: `httpation_request:method(Method, Request)`